File: src/availability-processor/src/events.go
package processor

import "time"

type CaregiverPermanentUnavailabilityEvent struct {
	Id            string    // Unique identifier for the permanent unavailability event
	TenantId      string    // Unique identifier for the tenant (care agency)
	CaregiverId   string    // Unique identifier for the caregiver
	EffectiveFrom time.Time // Time when the permanent unavailability starts
}

type CaregiverAbsenceBookedEvent struct {
	Id          string    // Unique identifier for the absence event
	TenantId    string    // Unique identifier for the tenant (care agency)
	CaregiverId string    // Unique identifier for the caregiver
	StartTime   time.Time // Start time of the absence
	EndTime     time.Time // End time of the absence
}

File: src/availability-processor/typescript/src/events.ts
export interface CaregiverPermanentUnavailabilityEvent {
  /** Unique identifier for the permanent unavailability event */
  id: string;
  /** Unique identifier for the tenant (care agency) */
  tenantId: string;
  /** Unique identifier for the caregiver */
  caregiverId: string;
  /** Time when the permanent unavailability starts */
  effectiveFrom: Date;
}

export interface CaregiverAbsenceBookedEvent {
  /** Unique identifier for the absence event */
  id: string;
  /** Unique identifier for the tenant (care agency) */
  tenantId: string;
  /** Unique identifier for the caregiver */
  caregiverId: string;
  /** Start time of the absence */
  startTime: Date;
  /** End time of the absence */
  endTime: Date;
}
